spyder/utils/conda.py

```diff
@@ -130,14 +133,24 @@ def get_conda_env_path(pyexec, quote=False):
     return conda_env
 
 
-def find_conda():
-    """Find conda executable."""
+def find_conda(pyexec=None):
+    """
+    Find conda executable.
+
+    `pyexec` is a python executable, the relative location from which to
+    attempt to locate a conda executable.
+    """
     # First try the environment variables
     conda = os.environ.get('CONDA_EXE') or os.environ.get('MAMBA_EXE')
     if conda is None:
         # Try searching for the executable
         conda_exec = 'conda.bat' if WINDOWS else 'conda'
```

Comment:
For Spyder 6 we need to expand this to also include `mamba` and `micromamba`. Otherwise, we won't be able to activate envs when Spyder is installed, for instance, in a Mambaforge installation that is present in a non-standard location (I think it's fine to leave this as it is for Spyder 5).

Could you take care of that @mrclary?

Reply:
Yep, I can take a look at it.